src/main/java/net/snowflake/ingest/SimpleIngestManager.java

```diff
@@ -566,6 +568,52 @@ public IngestResponse ingestFiles(
     return ServiceResponseHandler.unmarshallIngestResponse(response, requestId);
   }
 
+  /**
+   * ingestFiles With Client Info - synchronously sends a request to the ingest service to enqueue
+   * these files along with clientSequencer and offSetToken.
+   *
+   * <p>OffsetToken will be atomically persisted on server(Snowflake) side along with files if the
+   * clientSequencer added in this request matches with what Snowflake currently has.
+   *
+   * <p>If clientSequencers doesnt match, 400 response code is sent back and no files will be added.
+   *
+   * @param files - list of wrappers around filenames and sizes
+   * @param requestId - a requestId that we'll use to label - if null, we generate one for the user
+   * @param showSkippedFiles - a flag which returns the files that were skipped when set to true.
+   * @param clientInfo - clientSequencer and offsetToken to pass along with files.
+   * @return an insert response from the server
+   * @throws URISyntaxException - if the provided account name was illegal and caused a URI
+   *     construction failure
+   * @throws IOException - if we have some other network failure
+   * @throws IngestResponseException - if snowflake encountered error during ingest
+   * @throws BackOffException - if we have a 503 response
+   */
+  public IngestResponse ingestFiles(
+      List<StagedFileWrapper> files,
+      UUID requestId,
+      boolean showSkippedFiles,
+      InsertFilesClientInfo clientInfo)
+      throws URISyntaxException, IOException, IngestResponseException, BackOffException {
+
+    // the request id we want to send with this payload
+    if (requestId == null || requestId.toString().isEmpty()) {
+      requestId = UUID.randomUUID();
+    }
+
+    if (clientInfo == null) {
+      return ingestFiles(files, requestId, showSkippedFiles);
```

Comment:
From a readability standpoint, it will be good if this logic is inline and the previous method calls this with a null value for InsertFilesClientInfo

Reply:
Makes sense. The new one is more readable.